ansible_collections/arista/avd/plugins/action/anta_runner.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+
+from ansible.errors import AnsibleActionFail
+from ansible.plugins.action import ActionBase
+from anta.cli.utils import setup_logging
+from anta.inventory.models import InventoryDeviceHttpApi
+from anta.loader import parse_catalog
+from anta.result_manager import ResultManager
+from yaml import safe_load
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils import update_catalog
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        play_context = self._play_context
+        connection = self._connection.httpapi
+
+        # Need to set the right become_context to the _connection.httpapi object
+        connection.set_become(play_context)
```

Comment:
Since you moved the connection integration into ANTA, this should also be covered in there. Otherwise someone running anta outside of AVD would be missing this.